src/ray/core_worker/core_worker.cc

```diff
@@ -2183,7 +2224,7 @@ std::shared_ptr<rpc::RuntimeEnvInfo> CoreWorker::OverrideTaskOrActorRuntimeEnvIn
     auto job_serialized_runtime_env =
         worker_context_.GetCurrentJobConfig().runtime_env_info().serialized_runtime_env();
     if (!IsRuntimeEnvEmpty(job_serialized_runtime_env)) {
-      parent = std::make_shared<json>(json::parse(job_serialized_runtime_env));
+      parent = GetCachedJsonRuntimeEnvOrParse(job_serialized_runtime_env);
```

Comment:
some thoughts:

1. `parent` is const per job, so instead of parsing everytime (master) or asking a cache every time (this PR), we can just define a new, per-job `WorkerContext::GetCurrentJobRuntimeEnv()`, just like the per-task `WorkerContext::GetCurrentRuntimeEnv()`, which parses ahead of the time and it kept throughout the job.
2. same thing to the copy of parent_runtime_env_info
3. the cache in this PR only caches the json/pb parsing part, but now the runtime env merging part. Considering we always use the merged rtenv, we can do caching in a job level for merged rtenvs. How it works: make a LRU cache for std::shared_ptr<rpc::RuntimeEnvInfo>; on job assignment clear it; on call to OverrideTaskOrActorRuntimeEnvInfo, first consult the LRU cache.

Reply:
we can also discuss this offline

Reply:
Ok, it seems there're two things I need to cache:
- Job-wise parsed json
- Merged pb runtime env info, key-ed by serialized runtime env

Reply:
I think the overall proposal makes sense to me, but it *might* adds runtime overhead due to the enlarged scoped of cached value execution.
The ideal implementation might be having another sharded cache to reduce critical section, which I would prefer to leave it to next PR.

Reply:
Implementation updated.